aiosmtpd/smtp.py

```diff
@@ -434,7 +434,11 @@ def smtp_MAIL(self, arg):
             yield from self.push(syntaxerr)
             return
         arg = self._strip_command_keyword('FROM:', arg)
-        address, params = self._getaddr(arg)
+        try:
+            address, params = self._getaddr(arg)
```

Comment:
It might make more sense to catch this in `_getaddr()` and also add a catch for `HeaderParseError` so that we can move the try/except out of `smtp_VRFY()`.  Here's my thinking:

If `_getaddr()` is defined to either return a valid, non-empty address or `(None, None)` in the case of failure, then the three sites that call `_getaddr()` can all look like:

```python
address, params = self._getaddr(arg)
if address is None:
    yield from self.push(syntaxerror)
```
(or close since the error will be a little different for `smtp_VRFY()`)